lib/piper/permissions/ast/chars/rule.ex

```diff
@@ -3,7 +3,11 @@ defimpl String.Chars, for: Piper.Permissions.Ast.Rule do
   alias Piper.Permissions.Ast
 
   def to_string(%Ast.Rule{command_selector: cs, permission_selector: ps}) do
-    "when #{cs} must have #{ps}"
+    if "#{ps}" == "allow" do
```

Comment:
Might be nice to have this in a separate function definition. This is probably fine though.